doc/crypto/api/keys/management.rst

```diff
@@ -596,3 +516,162 @@ Key export
+
+            For HMAC keys that are shorter than, or equal in size to, the block size of ``hash_alg``, the format is the raw bytes of the key.
+
+            For HMAC keys that are longer than the block size of ``hash_alg``, the format is an :scterm:`implementation defined` choice between the following formats:
+
+            1.  The raw bytes of the key.
+            #.  The raw bytes of the hash of the key, using ``hash_alg``.
+
+    *   -   Other symmetric keys
+
+            *   AES
+            *   ARC4
+            *   ARIA
+            *   CAMELLIA
+            *   ChaCha20
+            *   SM4
+            *   Secrets for derivation
+            *   Password hashes
+
+        -   `PSA_KEY_TYPE_AES`
+
+            `PSA_KEY_TYPE_ARC4`
+
+            `PSA_KEY_TYPE_ARIA`
+
+            `PSA_KEY_TYPE_CAMELLIA`
+
+            `PSA_KEY_TYPE_CHACHA20`
+
+            `PSA_KEY_TYPE_SM4`
+
+            `PSA_KEY_TYPE_DERIVE`
+
+            `PSA_KEY_TYPE_PASSWORD_HASH`
+
+            The key data is the raw bytes of the key.
+
+    *   -   RSA key pair
+        -   `PSA_KEY_TYPE_RSA_KEY_PAIR`
+
+            The key data is the non-encrypted DER encoding of the representation defined by in :RFC-title:`8017` as ``RSAPrivateKey``, version ``0``.
+
+            .. code-block:: none
+
+                RSAPrivateKey ::= SEQUENCE {
+                    version             INTEGER,  -- must be 0
+                    modulus             INTEGER,  -- n
+                    publicExponent      INTEGER,  -- e
+                    privateExponent     INTEGER,  -- d
+                    prime1              INTEGER,  -- p
+                    prime2              INTEGER,  -- q
+                    exponent1           INTEGER,  -- d mod (p-1)
+                    exponent2           INTEGER,  -- d mod (q-1)
+                    coefficient         INTEGER,  -- (inverse of q) mod p
+                }
+
+            .. note::
+
+                Although it is possible to define an RSA key pair or private key using a subset of these elements, the output from `psa_export_key()` for an RSA key pair must include all of these elements.
+
+    *   -   RSA public key
+        -   `PSA_KEY_TYPE_RSA_PUBLIC_KEY`
+
+            The key data is the DER encoding of the representation defined by :RFC-title:`3279#2.3.1` as ``RSAPublicKey``.
+
+            .. code-block:: none
+
+                RSAPublicKey ::= SEQUENCE {
+                    modulus            INTEGER,    -- n
+                    publicExponent     INTEGER  }  -- e
+
+    *   -   Weierstrass Elliptic curve key pair
+        -   :code:`PSA_KEY_TYPE_ECC_KEY_PAIR(ecc_family)`, where ``ecc_family`` designates a Weierstrass curve family.
+
+            The key data is the content of the ``privateKey`` field of the ``ECPrivateKey`` format defined by :RFC-title:`5915`.
+
+            This is a ``ceiling(m/8)``-byte string in big-endian order where ``m`` is the key size in bits.
```

Comment:
I can see that almost all of the ECC key formats specify an endianness when the key is just a string representation of a scalar integer. The only one that does not specify endianness is the Twisted Edwards curve family keys - I can check if these are scalars, and should be updated to match.

Reply:
Ed25519 and Ed448 private keys are not numbers (they are a seed for a very simple key derivation function, which the signature and private-to-public algorithms use to calculate the number that is the “mathematical” private key). Hence they have no endianness.

Ed25519 and Ed448 public keys consist of an integer (the Y coordinate) written in little-endian format, plus an extra bit which is the sign of the X coordinate, plus 1–7 bits to pad to a whole number of octets. I think it's ok for the specification to keep just referencing [RFC 8032](https://www.rfc-editor.org/rfc/rfc8032#section-3.1) instead of inlining the full description.

Reply:
@paul-elliott-arm - Are there other instances that we have missed here?